### PR TITLE
Fix NullPointerException in detach network from cluster

### PR DESCRIPTION
**Pull request by MatheMatrix**
The vrouter vm do not have a defaultL3Uuid cause this exception.

For zstackio/issues#2280
Partial: zstackio/issues#1689

### CODE REVIEW

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java [hunk 1]
```diff
@@ -2955,13 +2955,18 @@ private void removeStaticIp() {
...
-                if (!self.getDefaultL3NetworkUuid().equals(nic.getL3NetworkUuid())) {
+                String defaultL3Uuid = dbf.findByUuid(spec.getVmInventory().getUuid(), VmInstanceVO.class).getDefaultL3NetworkUuid();
+                if (defaultL3Uuid != null && defaultL3Uuid.equals(nic.getL3NetworkUuid())) {
                     return;
                 }
 
                 final VmInstanceInventory vm = getSelfInventory();
                 final String previousDefaultL3 = vm.getDefaultL3NetworkUuid();
 
+                if (vm.getType().contentEquals("ApplianceVm")) {
```
Comment (zxwing) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: 不能这样hardcode的写

Comment (MatheMatrix) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: done

**Comment by MatheMatrix** [comment]
updated,
the new commit contains two fixes, and add portforwarding to test

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java [hunk 2]
```diff
@@ -2955,13 +2955,17 @@ private void removeStaticIp() {
...
             private void selectDefaultL3() {
-                if (!self.getDefaultL3NetworkUuid().equals(nic.getL3NetworkUuid())) {
+                if (self.getDefaultL3NetworkUuid() != null && self.getDefaultL3NetworkUuid().equals(nic.getL3NetworkUuid())) {
                     return;
                 }
 
                 final VmInstanceInventory vm = getSelfInventory();
                 final String previousDefaultL3 = vm.getDefaultL3NetworkUuid();
 
+                if (!vm.getType().contentEquals(VmInstanceConstant.USER_VM_TYPE)) {
```
Comment (zxwing) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: user override to fix this

**Pull request merged**